torch/nn/parallel/data_parallel.py

```diff
@@ -78,17 +78,21 @@ def _to_cuda(obj):
 
         replicas = self.replicate(self.module, self.device_ids)
         scattered = self.scatter(inputs, self.device_ids)
-
+        used_gpus = len(scattered)  # The last GPU might not be used. For example, input of size 5, on 4 GPUs
         gpu_dicts = None
         if kwargs:
             scatter_kwargs = {}
             for key in kwargs.keys():
                 scatter_kwargs[key] = self.scatter(
                     _to_cuda(kwargs[key]), self.device_ids)
-            gpu_dicts = tuple(
-                {key: values[i] for key, values in scatter_kwargs.items()}
-                for i in self.device_ids
-            )
+
+            gpu_dicts = tuple()
+            for i in range(used_gpus):
```

Comment:
Wouldn't it be simpler to do that in the loop above, without the use of scatter_kwargs? Just put the scatter results directly into the right gpu dicts